cpp/src/nvgraph/matrix.cu

```diff
@@ -216,22 +216,6 @@ namespace nvgraph {
     Cusparse::set_pointer_mode_host();
   }
 
-  /// Constructor for CSR matrix class
-  /** @param G Weighted graph in CSR format
-   */
-  template <typename IndexType_, typename ValueType_>
-  CsrMatrix<IndexType_,ValueType_>
-  ::CsrMatrix(  ValuedCsrGraph<IndexType_,ValueType_> & G, const cusparseMatDescr_t _descrA)
-    : Matrix<IndexType_,ValueType_>(G.get_num_vertices(), G.get_num_vertices()),
-      trans(false), sym(false),
-      nnz(G.get_num_edges()),
-      descrA(_descrA), 
-      csrValA(G.get_raw_values()),
-      csrRowPtrA(G.get_raw_row_offsets()),
-      csrColIndA(G.get_raw_column_indices()) {
-    Cusparse::set_pointer_mode_host();
-  }
-
   /// Destructor for CSR matrix class
   template <typename IndexType_, typename ValueType_>
```

Comment:
Afaik this is really just the matrix class used for the spectral clustering feature. We could add a `spectral_` prefix to the file name for now and try to see if it can go away in the long run.

Reply:
I can rename this.  I actually looked into getting rid of it entirely, but that was going to be more work.  I imagine we'll want to rework these spectral techniques to use GraphBLAS once it's ready.